bluepymm/prepare_combos/parse_files.py

```diff
@@ -128,10 +128,10 @@ def read_mm_recipe_yaml(recipe_filename):
     mecombos = pandas.DataFrame(columns=["layer", "fullmtype", "etype"])
     for region in recipe['neurons']:
         for etype in region['traits']['etype'].keys():
-            n_combos = len(mecombos)
-            mecombos.loc[n_combos, 'layer'] = region['traits']['layer']
-            mecombos.loc[n_combos, 'fullmtype'] = region['traits']['mtype']
-            mecombos.loc[n_combos, 'etype'] = etype
+            n = len(mecombos)
+            mecombos.loc[n, 'layer'] = str(region['traits']['layer'])
+            mecombos.loc[n, 'fullmtype'] = str(region['traits']['mtype'])
+            mecombos.loc[n, 'etype'] = str(etype)
```

Comment:
So I assume this needs to be added to make it work with the latest circuit?
@anilbey we don't have to block this PR for it, but can we change the relevant to make it fail in the same way as the circuit did.

Reply:
It was failing here: https://github.com/BlueBrain/BluePyMM/blob/master/bluepymm/prepare_combos/create_mm_sqlite.py#L363
the outputed dataframe had Nan on the column 'etype', which means that the merge operation did not associate the layer/mtype between dataframes (object vs str)

Reply:
I didn't understand. The str conversion is only to make the yaml parser produce the same output as the xml parser. 
Why is the circuit failing?

Reply:
I think werner means that this bug prevents us to run MM on the new release of clone cells, but apical as well.

Reply:
> It was failing here: https://github.com/BlueBrain/BluePyMM/blob/master/bluepymm/prepare_combos/create_mm_sqlite.py#L363
> the outputed dataframe had Nan on the column 'etype', which means that the merge operation did not associate the layer/mtype between dataframes (object vs str)

I didn't see this when I wrote the above comment. 
Ok I was expecting this because the yaml.safe_load is assigning the object type to some variables.